x/stake/keeper/validator.go

```diff
@@ -129,11 +139,7 @@ func (k Keeper) AddValidatorTokensAndShares(ctx sdk.Context, validator types.Val
 	k.DeleteValidatorByPowerIndex(ctx, validator)
 	pool := k.GetPool(ctx)
 	validator, pool, addedShares = validator.AddTokensFromDel(pool, tokensToAdd)
-	// increment the intra-tx counter
-	// in case of a conflict, the validator which least recently changed power takes precedence
-	counter := k.GetIntraTxCounter(ctx)
-	validator.BondIntraTxCounter = counter
-	k.SetIntraTxCounter(ctx, counter+1)
+	k.BumpValidatorBondHeightAndCounter(ctx, &validator)
```

Comment:
Not sure this makes sense - In the old code, the bond height remains the same (it stays old) whereas not we are _also_ updating the bond height, which means that even if you're an old validator, your "precedence" that you get from having an old bond height is effectively erased every time anyone bonds to you... As I understand it the bond height should _only_ ever get updated when you're actually bonding (aka in the master update function `ApplyAndReturnValidatorSetUpdates` - which is currently happening)

Reply:
Semantically yes, and that's the problem with this... I'm more concerned about not wiping out a validator (not even in the cliff) by ending up with the same stake amount, which is a grief'ing factor, so wanted to ensure that keys were unique.  Solving this now.